container-images/tools
# add tests + standard test invocation

- [ ] TODO: make the playbook executable on remote host (workflow: provision a VM in openstack, execute the playbook against the fresh VM -- `hosts: something`, not localhost)

### Requirements

0. Familairty with standard test interface:
    * https://fedoraproject.org/wiki/CI/Tests
    * https://fedoraproject.org/wiki/Changes/InvokingTests
1. Ansible, distgen (`dnf install ansible distgen`)
2. docker running
3. Requires these changes: https://pagure.io/standard-test-roles/pull-request/121
   You can just clone those changes locally and then point ansible to look for those roles like this:
   ```
   $ tail -n 1 ./tests/ansible.cfg
   roles_path = </path/to/the/repo>/roles
   ```

### How to test?

This will invoke the tests in your current environment:
```
$ make check

...

TASK [standard-test-basic : Execute tests] ********************************
changed: [localhost] => (item=integration)

TASK [standard-test-basic : Check the results] ****************************
changed: [localhost]

TASK [standard-test-basic : include_role] *********************************

TASK [str-common : Pull out the logs from test environment to test runner]
changed: [localhost]

TASK [delete the temp dir] ************************************************
changed: [localhost]

PLAY RECAP ****************************************************************
localhost                  : ok=20   changed=8    unreachable=0    failed=0

make[1]: Leaving directory '/home/tt/g/tools/tests'
```

No errors nor failures = successful run.


There is also possibility to invoke tests inside a freshly provisioned VM:
```
$ make check-in-vm
```

Logs from the tests are stored over here:
```
$ cat tests/artifacts/test.integration.log
============================= test session starts ==============================
platform linux -- Python 3.6.4, pytest-3.2.3, py-1.5.2, pluggy-0.4.0 -- /usr/bin/python3
cachedir: .cache
rootdir: /tmp/ansible.mo9bas/integration, inifile:
plugins: cov-2.5.1, catchlog-1.2.2, celery-4.0.2
collecting ... collected 8 items

test_container.py::TestContainer::test_ethtool PASSED
test_container.py::TestContainer::test_netstat PASSED
test_container.py::TestContainer::test_ss PASSED
test_container.py::TestContainer::test_pstack PASSED
test_container.py::TestContainer::test_nstat PASSED
test_container.py::TestContainer::test_numastat PASSED
test_container.py::TestContainer::test_pmap PASSED
test_container.py::TestContainer::test_strace PASSED

=========================== 8 passed in 4.33 seconds ===========================
```
